spec/classes/foreman_plugin_discovery_spec.rb

```diff
@@ -49,12 +46,9 @@
       should contain_foreman__plugin('discovery')
     end
 
-    it 'should not download and install kernel file' do
-      should_not contain_foreman__remote_file("/var/lib/tftpboot/boot/foreman-discovery-image-latest.el6.iso-vmlinuz")
-    end
-
-    it 'should not download and install initrd file' do
-      should_not contain_foreman__remote_file("/var/lib/tftpboot/boot/foreman-discovery-image-latest.el6.iso-img")
+    it 'should not download and install tarball' do
+      should_not contain_foreman__remote_file("/var/lib/tftpboot/boot/fdi-image-latest.tar").
+        with_remote_location('http://yum.theforeman.org/discovery/releases/latest/fdi-image-latest.tar')
```

Comment:
best to remove this line, so you're just checking for the resource not to exist, rather than the resource with a specific parameter